src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -574,5 +595,35 @@ public int getWascId() {
         		AlertThreshold.MEDIUM};
     }
 
+	@Override
+	public void scan(HttpMessage msg, NameValuePair originalParam) {
+		if (originalParam.getType() != NameValuePair.TYPE_QUERY_STRING) {
+			return; // Exit if it isn't a URL param
+		}
+		super.scan(msg, originalParam);
+	}
+
+	@Override
+	public void scan(HttpMessage msg, String paramName, String attack) {
+		Stats.incCounter("domxss.scan.count");
+
+		ArrayList<WebDriverWrapper> drivers = new ArrayList<WebDriverWrapper>();
+
+		WebDriverWrapper fxDriver;
+		fxDriver = this.getFirefoxDriver();
+		drivers.add(fxDriver);
+
+		for (String attackVector : PARAM_ATTACK_STRINGS) {
+			if (scan(drivers, paramName, attackVector, "")) {
+				if (!Plugin.AlertThreshold.LOW.equals(this.getAlertThreshold())) {
+					// Only report one issue per URL unless
+					// Alert threshold is LOW
+					break;
+				}
+			}
+		}
+		this.returnFirefoxDriver(fxDriver);
```

Comment:
It's not necessary (it will be returned in the finally block) and `fxDriver` would no longer be needed.

Reply:
I had to restore this, if it wasn't there then Firefox windows would keep opening.....

Reply:
OK, need to check that, we might as well share the same driver (obtained in the scan method) instead of getting them again here.